packages/core/src/html/SiteLinkManager.ts

```diff
@@ -69,4 +74,48 @@ export class SiteLinkManager {
     this._addToCollection(resourcePath, cwf);
     return 'Intralink collected to be validated later';
   }
+
+  maintainFilePathToHashesMap(node: MbNode, cwf: string, forceWrite: string = '') {
+    if (!this.config.intrasiteLinkValidation.enabled) {
+      return;
+    }
+    const path = cwf.substring(this.config.rootPath.length);
+    if (!this.filePathToHashesMap.has(path)) {
+      this.filePathToHashesMap.set(path, new Set());
+    }
+    if (forceWrite !== '') {
+      this.filePathToHashesMap.get(path)!.add(forceWrite);
+    } else if (node.attribs!.id) {
+      this.filePathToHashesMap.get(path)!.add(node.attribs!.id);
+    }
+  }
+
```

Comment:
Hmmm I see that this is mostly used for testing purposes. Should this be moved into test file instead?

Reply:
I want argue for the current code here;

To test the internal state of the siteLinkManager, we need to either 1 write getter for the filePathToHashesMap, 2 write comparator inside the SiteLinkManager 3 add print method.  I think 3, the current way is the least intrusive (and most helpful when I do debugging with console.log)

Reply:
@yiwen101 

I think I am okay with this method if you really cannot find a better way but you may need to rename it since it does not really print but return the string of the hashmap.

I think it is possible to create a class that extends from SiteLinkManager in the test file right that has the getter? Do you think this will work and fix the issue?

Reply:
@EltonGohJH 
Thank you so much for your reply, I was not aware that I can do testing like this in typescript before! This is valuable learning point for me;

I have deleted this method in the latest commit, and created a class extending the siteLinkManager as instructed by you. It works well!